### PR TITLE
GH72 XLIB=Off vktraceviewer

### PR DESCRIPTION
**Pull request by davidhubbard**
Here is a potential fix for the XLIB=Off issue (#72).

@ardaLunarG can you please take a look?

@karl-lunarg suggested a pull request that builds vktraceviewer with XLIB Off and VKTRACEVIEWER On. I can build vktraceviewer (I do not have XLIB installed) with the following:

```
git checkout ...
./update_external_sources.sh
cmake -H. -Bbuild -DCMAKE_BUILD_TYPE=Debug -DBUILD_WSI_XLIB_SUPPORT=Off -DBUILD_VKTRACEVIEWER=On
cd build
make
```

### CODE REVIEW

File: vulkan.py [hunk 1]
```diff
@@ -159,11 +159,12 @@ def __repr__(self):
                 (self.ret, self.name, param_str)
 
 class Extension(object):
-    def __init__(self, name, headers, objects, protos):
+    def __init__(self, name, headers, objects, protos, ifdef = None):
```
Comment (ardaLunarG) [inline vulkan.py]: This file "vulkan.py" belongs to LVL (https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers)
This patch needs to split and commit this file to LVL and vk-vtgenerate.py changes to VulkanTools

**Comment by ardaLunarG** [comment]
David, I will handle this split patch and commit if you are OK with it, I will take some time as LVL changes needs to be merged after commit so VulkanTools compiles fine.

**Comment by davidhubbard** [comment]
Hi Arda, I am OK with that. :)

**Comment by ardaLunarG** [comment]
Cherry picked to master

**Pull request closed without merging**